base/uk.ac.stfc.isis.ibex.scriptgenerator/defined_actions/action_loader.py

```diff
@@ -0,0 +1,72 @@
+from py4j.clientserver import ClientServer, JavaParameters, PythonParameters
+from py4j.java_collections import ListConverter
+import inspect
+import argparse
+from action_interface import Action
+import os
+import sys
+
+
```

Comment:
There are multiple times where dict is used as a type hint. This should not be done, we should import Dict from the typing module and use that.

Reply:
We should also attempt to provide the best possible type hinting with Dict by specifying key and value types if possible e.g. Dict[str, TypeVar[T]] or Mapping[str, TypeVar[T]]